lib/IDE/SwiftSourceDocInfo.cpp

```diff
@@ -581,7 +581,7 @@ std::pair<bool, Pattern*> NameMatcher::walkToPatternPre(Pattern *P) {
   if (isDone() || shouldSkip(P->getSourceRange()))
     return std::make_pair(false, P);
 
-  tryResolve(ASTWalker::ParentTy(P), P->getLoc());
+  tryResolve(ASTWalker::ParentTy(P), P->getStartLoc());
```

Comment:
@nkcsgexi is this OK?

Reply:
I think this is fine for patterns. We need to be careful for decls though, sometimes `getLoc()` and `getStartLoc` point to different position IIRC.